plugins/de.cognicrypt.staticanalyzer/src/de/cognicrypt/staticanalyzer/markerresolution/QuickFixer.java

```diff
@@ -13,17 +17,33 @@
  * @author André Sonntag
  */
 public class QuickFixer implements IMarkerResolutionGenerator {
+	ArrayList<IMarkerResolution> quickFixes;
+	boolean toggle = true;
+
 	@Override
 	public IMarkerResolution[] getResolutions(final IMarker mk) {
 
+		quickFixes = new ArrayList<>();
 		String message = "";
+
 		try {
 			message = (String) mk.getAttribute(IMarker.MESSAGE);
-		}
-		catch (final CoreException e) {
+			quickFixes.add(new SuppressWarningFix(Constants.SUPPRESSWARNING_FIX + message));
+
+			if (toggle) {
+				if (mk.getAttribute("error") instanceof RequiredPredicateError) {

```

Comment:
This doesn't work. Marker attributes must be boolean, integer or String. Other stuff cannot be serialized to disk, and violates the IMarker.setAttribute(...) and getAttribute(...) API (see Javadoc).